src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -116,7 +116,7 @@ object ResourceMatcher {
     def portsMatchOpt: Option[PortsMatch] = new PortsMatcher(app, offer, selector).portsMatch
 
     def meetsAllConstraints: Boolean = {
-      lazy val tasks = runningTasks
+      lazy val tasks = runningTasks.filter(t => t.launched.map(_.appVersion == app.version).getOrElse(true))
```

Comment:
Every scaling change will change the appVersion, so a strict equals is wrong.
Also if the task is not launched, the default should be `false` not `true`.
Perhaps something like: `runningTasks.filter(_.launched.exists(_.appVersion >= app.versionInfo.lastConfigChangeVersion))` is better.